### DISCUSSION TITLE
Interpolate for Affine (pyvip)

### DISCUSSION THREAD

**Discussion by d5423197** [libvips/libvips#2936] [Q&A]
Hello Author,

I am trying to using pyvips affine instead of numpy rot90, I compared the result, and they are a bit different. I am guessing it might be caused by interpolation. I am using nearest for pyvips affine. But in my opinion, if the image is rotated by 90 degree, there should not be any interpolation involved. Please help, thanks

**Comment by jcupitt** [comment]
Hello, `affine` uses corner convention, and the `rot*` functions use centre convention.

You can use the `idx` and `idy` parameters to `affine` to flip to centre convention.

**Comment by d5423197** [comment]
Thanks for your reply. I am looking for some examples for the affine function. I have read the API but still could not fully understand.

**Comment by d5423197** [comment]
Hello there,

Based on your reply, I could not understand how to use idx and idy to affine to centre convention. Is there any blogs that introduces corner convention? I haven googled it but could not find some useful info.

**Comment by d5423197** [comment]
refer to Rotating center in affine #1023 and Rotation center for similarity transform ? #948, I have tried to add idx and idy as follows.

```python
arr = arr.affine(m, 
                 interpolate=pyvips.Interpolate.new("nearest"), 
                 background=[0, 0, 0, 0], 
                 idx=-c_[0],
                 idy=-c_[1],
                 odx=c_[0],
                 ody=c_[1])
```

where c is the center of the original image. I also checked the doc for affine which states "By default oarea is just large enough to cover the whole of the transformed input image." But in this case, if I did not specify the oarea, the output will be a black image. If I set orea as (0, 0, width, height). The output shape is not equal to the result that I get from numpy rot90.  

I am still trying to find some details about affine in libvips. Still could not fully understand the meaning of each parameter for affine.

**Comment by d5423197** [comment]
@jcupitt Hello author, any help will be appreciated.
